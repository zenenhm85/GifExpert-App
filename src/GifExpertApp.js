import React, { useState } from "react";
import { CategoryAdd } from "./components/CategoryAdd";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = ["Batman"] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <CategoryAdd setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
