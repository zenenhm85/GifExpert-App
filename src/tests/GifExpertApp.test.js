import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifExpertApp } from "../GifExpertApp";

describe('Tests for GifExpertApp', () => {
    test("should display the component correctly", () => {
       
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
      });    
      test("should display the component correctly, when items gifts are providers", () => {
       
        const categories = ["Spider", "Simpson"]
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
      }); 
})
