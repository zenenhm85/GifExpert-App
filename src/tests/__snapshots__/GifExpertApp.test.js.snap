// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifExpertApp should display the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
  </ol>
</Fragment>
`;

exports[`Tests for GifExpertApp should display the component correctly, when items gifts are providers 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Spider"
      key="Spider"
    />
    <GifGrid
      category="Simpson"
      key="Simpson"
    />
  </ol>
</Fragment>
`;
